### DISCUSSION TITLE
Qutebrowser instance becomes very laggy when another instance on a website with moving objects is placed off-screen

### DISCUSSION THREAD

**Discussion by leonsilicon** [qutebrowser/qutebrowser#5872] [Q&A] [answered]
**Version info**:
qutebrowser v1.14.0
Git commit: 
Backend: QtWebEngine (Chromium 80.0.3987.163)
Qt: 5.15.1

**Does the bug happen if you start with `--temp-basedir`?**: No

**Description**
When using a window manager like dwm (https://dwm.suckless.org/), when a qutebrowser window is opened on a website like giphy.com and then placed off screen (e.g. assigned an inactive tag in dwm), opening another instance on screen will cause it to lag heavily. However, when the other website (giphy.com) is brought back on-screen, the lag stops.

I originally experienced this issue on SoundCloud, and I suspected it had something to do with things moving in the browser, and so I tried with giphy.com and sure enough it became really laggy. This bug didn't seem to happen on google.com, though.

Edit: I was able to reproduce this bug on another website with moving objects: https://turbo.fish and it's also having the lag issues.

Edit 2: I just tried this on i3 (https://i3wm.org) and it doesn't seem to have this issue. I think this a dwm specific issue.

**How to reproduce**

1. Open a qutebrowser instance and visit giphy.com
2. Move it off-screen
3. Open another instance and try typing anything - it will be laggy
4. Move the giphy.com qutebrowser instance back on screen
5. Try typing anything again on the other instance - the lag will be gone

Here's a video I recorded of this issue:
https://streamable.com/6euhqe

**Comment by leonsilicon** [comment, marked as answer]
Ok, I just tried reproducing this on Brave and Chrome and they both also seem to display the same issue, so I don't think this issue is specific to qutebrowser. However, running qutebrowser with --temp-basedir does fix the lag, but I'm not sure why.

**Reply by The-Compiler** [reply]
I converted this to a discussion as it's clearly not an issue in qutebrowser then.

I assume without `--temp-basedir`, you have two windows running in the same qutebrowser instance (if you launch qutebrowser a second time, it just tells the running instance to open a new window).

If you launch twice with `--temp-basedir`, you actually get two separate instances. Can you try with `--temp-basedir` and `:open -w` inside qutebrowser, or with `--basedir /tmp/somedirectory` when launching twice (so that they end up with the same basedir, i.e. the same intance)?

**Reply by leonsilicon** [reply]
Yep, I just tried reproducing this with --temp-basedir and :open -w (and also separately with --basedir ~/temp), and they both exhibit the same lag issues.

**Reply by leonsilicon** [reply]
For what it's worth, I also just tried this on Firefox, and it seems like Firefox doesn't have the same lag issue as Brave and Chromium.

**Comment by leonsilicon** [comment]
I just tried reproducing this today with Brave and Chrome, and it seems like it's been fixed in Version 87.0.4280.88 of Chromium (which I'm using in both Brave and Chrome). I noticed Qutebrowser 1.14.1 is running version Chromium 83 and I can still reproduce the bug on Qutebrowser.

**Reply by The-Compiler** [reply]
> I noticed Qutebrowser 1.14.1 is running version Chromium 83

That entirely depends on what QtWebEngine version you're using it with - but the newest QtWebEngine release (5.15.2) is indeed based on Chromium 83. Looks like the next release might come with Chromium 87 though: [BASELINE: Update Chromium to 87.0.4280.89 (Icee7f7f1) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine-chromium/+/325268)

**Reply by leonsilicon** [reply]
Oh, thanks for the update! Hopefully the update to Chromium 87 will fix the bug once and for all :)

**Comment by mashrafdev** [comment]
this issue fixed in my machine by removing xf86-video-intel and in /etc/environment add this line ```LIBGL_DRI3_DISABLE=1``` to use dri 2, since dri 3 may cause issues.

https://wiki.archlinux.org/index.php/Intel_graphics#DRI3_issue
